src/modules/RA/DataSource/components/DataSourceTabbedForm.js

```diff
@@ -15,36 +15,41 @@ import {
 import DataSourceMainFields from './DataSourceMainFields';
 import DataSourceFileFields from './DataSourceFileFields';
 import DataSourceWMTSField from './DataSourceWMTSField';
+import DataSourceCSVFields from './DataSourceCSVFields';
 import FieldSample from '../../../../components/react-admin/FieldSample';
 import AttributeMessage from './AttributeMessage';
 import DataSourceDbFields from './DataSourceDbFields';
-import {
-  SQL,
-  GEOJSON,
-  SHP,
-  WMTS,
-  fieldTypeChoices,
-} from '..';
+import { SQL, GEOJSON, SHP, WMTS, CSV, fieldTypeChoices } from '..';
 
 const DataSourceTabbedForm = ({ translate: t, ...props }) => (
   <TabbedForm {...props}>
     <FormTab label="datasource.form.definition">
-
       <DataSourceMainFields />
 
       <FormDataConsumer>
         {({ formData: { _type: type } = {}, ...rest }) =>
-          type === WMTS && <DataSourceWMTSField {...rest} />}
+          type === WMTS && <DataSourceWMTSField {...rest} />
+        }
+      </FormDataConsumer>
+
+      <FormDataConsumer>
+        {({ formData: { _type: type } = {}, ...rest }) =>
+          [SHP, GEOJSON].includes(type) && (
+            <DataSourceFileFields {...rest} type={type} />
+          )
+        }
       </FormDataConsumer>
 
       <FormDataConsumer>
         {({ formData: { _type: type } = {}, ...rest }) =>
-          [SHP, GEOJSON].includes(type) && <DataSourceFileFields {...rest} type={type} />}
+          type === SQL && <DataSourceDbFields {...rest} />
+        }
       </FormDataConsumer>
 
       <FormDataConsumer>
         {({ formData: { _type: type } = {}, ...rest }) =>
-          type === SQL && <DataSourceDbFields {...rest} />}
+          [CSV].includes(type) && <DataSourceCSVFields {...rest} type={type} />
```

Comment:
```suggestion
          type === 'CSV' && <DataSourceCSVFields {...rest} type={type} />
```

Reply:
```suggestion
          type === CSV && <DataSourceCSVFields {...rest} type={type} />
```
Extraneous quotes ☝️